components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -496,6 +496,10 @@ std::vector<mojom::FeedItemV2Ptr> GenerateBlockFromContentGroups(
                 }
               }
 
+              if (metadata->publish_time < base::Time::Now() - base::Days(14)) {
```

Comment:
So we're adding a hard cutoff of two weeks? I think maybe we should also consider tweaking the half life of `GetPopRecency` to more strongly prefer more recent articles.

Additionally, if we're going to completely filter them out, I think we'd be better off doing it from the `GenerateAllFeed` method, so the source articles don't contain anything old (might be slightly better for early aborts too).

Reply:
I tweaked the weighting function, so now the pop score is a factor multiplying the other two. Before very old article weren't discounted because the pop score was added. Now, if the pop score is 0 or very low, so will the weight of those articles. 

I think we still need a filtering function because there might entire feeds with only old content (stale feeds). If we eliminate all articles from a publisher, then we should also prevent that empty publisher to be sampled as a content group. But it's all kind of tricky in the current setup. In general, I agree we should discard old articles as soon as possible in the process

Reply:
but right now, if there's an empty publisher that will result in a failed generation

Reply:
Yeah, this will fail if a publisher only has articles older than 2 weeks too - I dunno if its a good idea to merge as is - I have a lot of direct feeds which aren't updated very regularly so I'd expect this to terminate feed generation extremely quickly.

I think we might have to stop kicking the can down the road on fixing the approach :/

Reply:
removed this for now. It may still show old articles but at least the time-weighting relative to the chosen feed will be working. We should fix this approach asap and before doing any other changes to the ranking algo.